### PR TITLE
Frontier machine enviroment optimizations

### PR DESCRIPTION
**Pull request by sarats**
Just opening a PR to see what integrating this PR would entail.

Other than machine files, the Scorpio submodule and changes to 
`components/elm/src/biogeophys/BandDiagonalMod.F90` are the other major bits.

Should switch to cli115 instead of cli115-ecp as needed. May also need Scorpio update for ADIOS output.

### CODE REVIEW

**Comment by sarats** [comment]
@AaronDonahue Did you get a chance to run a CESS test using these machine settings?

**Review by Gdakgdak22 — APPROVED** [review]

File: components/elm/src/biogeophys/BandDiagonalMod.F90 [hunk 1]
```diff
@@ -218,4 +218,1891 @@ subroutine BandDiagonal(bounds, lbj, ubj, jtop, jbot, numf, filter, nband, b, r,
 
   end subroutine BandDiagonal
 
+
+!! reference impl
+!! from https://netlib.org/cgi-bin/netlibfiles.txt?format=txt&filename=/lapack/lapack_routine/dgbsv.f
+!! license https://netlib.org/lapack/LICENSE.txt
+
```
[the remark this answers is not in the document]
Comment (ndkeen) [reply components/elm/src/biogeophys/BandDiagonalMod.F90:225]: This is what I thought would be better

**Comment by sarats** [comment]
Reverted Land BandDiagonal changes and Scorpio module update. 
Just FYI, rebase is causing many conflicts, so reverted these specific commits using `git revert --strategy resolve commit`. 

Left https://github.com/E3SM-Project/scream/pull/2503/commits/4dbb07fe1c31ed0ea20c7511bacde6be80a33234 as it just reduces opt-level for the file. 
```
[sarat@login13 scream ]$ git revert --strategy resolve 4dbb07fe1c31ed0ea20c7511bacde6be80a33234
CONFLICT (modify/delete): cime_config/machines/Depends.frontier-scream-gpu.crayclang-scream.cmake deleted in parent of 4dbb07fe1c (custom flags for banddiag file) and modified in HEAD. Version HEAD of cime_config/machines/Depends.frontier-scream-gpu.crayclang-scream.cmake left in tree.
```

**Comment by sarats** [comment]
This includes a couple of commits applied as a patch that were since then merged into master in https://github.com/E3SM-Project/scream/pull/2431

**Comment by sarats** [comment]
Btw, I don't know about the provenance of this 'inout' issue in ELM.
https://github.com/E3SM-Project/scream/pull/2503/commits/95729a37a1cdd987c375129a13bc4c8a5326a941

Is this needed or should be reverted?

**Comment by AaronDonahue** [comment]
> @AaronDonahue Did you get a chance to run a CESS test using these machine settings?

@sarats , I have not.  Although I think either @ndkeen or @crterai would be better suited to give this branch a try on `frontier`.

**Comment by AaronDonahue** [comment]
>but could @AaronDonahue double check if they are the expected baseline diffs?

@ambrad , I don't see any reason in this PR that we should see DIFF's on `mappy`.

**Comment by crterai** [comment]
I ran a short ne1024 simulation where I merged the Frontier machine environment on top of the machines/frontier branch and it ran into the following crash pasted below 1.5 days (1479 steps) in to simulation. I've resubmitted to see if it's reproducible. 
The branch used is: https://github.com/E3SM-Project/scream/commits/machines/frontier_w_machenv_opt
And the git hash: `c407a49af9` 
Run directory: `/lustre/orion/cli115/proj-shared/terai/e3sm_scratch/cess-v2test_baselineForPerf.ne1024pg2_ne1024pg2.F2010-SCREAMv1.20230909/run/`
Run directory for baseline: `/lustre/orion/cli115/proj-shared/terai/e3sm_scratch/cess-v2test_baselineForPerf.ne1024pg2_ne1024pg2.F2010-SCREAMv1.20230901/run/`
Error message:
```
 7613: WARNING:CAAR: dp3d too small. k=30, dp3d(k)=43.483604, dp0=379.881287
 7613: WARNING:CAAR: dp3d too small. k=30, dp3d(k)=43.483604, dp0=379.881287
 7613: WARNING:CAAR: dp3d too small. k=30, dp3d(k)=43.483604, dp0=379.881287
 7613: WARNING:CAAR: dp3d too small. k=30, dp3d(k)=43.483604, dp0=379.881287
    0: bfbhash>           1476  2f98794acb30270 (Hommexx)
 7613:
 7613:  FAIL:
 7613: false
 7613: /autofs/nccs-svm1_home1/terai/SCREAM/code_w_perf_fix/components/eamxx/src/share/atm_process/atmosphere_process.cpp:442
 7613: Error! Failed post-condition property check (cannot be repaired).
 7613:   - Atmosphere process name: homme
 7613:   - Property check name: T_mid within interval [100, 500]
 7613:   - Atmosphere process MPI Rank: 7613
 7613:   - Message: Check failed.
 7613:   - check name: T_mid within interval [100, 500]
 7613:   - field id: T_mid[Physics PG2] <double:ncol,lev>(1536,128) [K]
 7613:   - minimum:
 7613:     - value: -1264.52
 7613:     - indices (w/ global column index): (24011477,126)
 7613:     - lat/lon: (56.2953, 236.449)
 7613:     - additional data (w/ local column index):
 7613:
...
```

**Comment by PeterCaldwell** [comment]
Yikes, it sounds like Sarat's performance branch causes the model to go unstable after 1.5 days. Both CAAR dp3d too small and T>500K are not what we want to see. Perhaps we should retreat to just rocm5.1?

**Comment by sarats** [comment]
I'm not aware of anything that should change numerics in my branch. It applies O2 for CXX explicitly which I presume is okay and is already default in EKAT and the ice files don't have opt on. Anyway, would be curious to see  the root cause of any numerical stability.

**Pull request closed without merging**